core/controllers/reader.py

```diff
@@ -421,6 +421,66 @@ class AnswerSubmittedEventHandler(base.BaseHandler):
     """Tracks a learner submitting an answer."""
 
     REQUIRE_PAYLOAD_CSRF_CHECK = False
+    URL_PATH_ARGS_SCHEMAS = {
+        'exploration_id': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': constants.ENTITY_ID_REGEX
+                }]
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'session_id': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'old_state_name': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'answer': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'client_time_spent_in_secs': {
+                'schema': {
+                    'type': 'int'
+                }
+            },
+            'answer_group_index': {
+                'schema': {
+                    'type': 'int'
+                }
+            },
+            'rule_spec_index': {
+                'schema': {
+                    'type': 'int'
+                }
+            },
+            'version': {
+                'schema': {
+                    'type': 'int',
+                    'validators': [{
+                        'id': 'is_at_least',
+                        'min_value': 1
+                    }]
+                },
+                'default_value': None
+            },
+            'classification_categorization': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            }
+        }
+    }
```

Comment:
You need to use `self.normalized_payload` instead of `self.payload` down below. Also, you can modify some of the code down below since you introduced some of the rules here.

Reply:
I made changes for self.normalized_payload, but I didn't really understand which part can be removed. The version check is still necessary, right? What else could be removed?

Reply:
Hey, after pushing this change, the frontend Karma test is failing and I am unable to identify the issue. Could you help with that, please?

Reply:
The frontend tests error is probably just a flake.

Reply:
Okay